### PR TITLE
Update master-node-communication.md

### PR DESCRIPTION
**Pull request by SrinivasChilveri**
Not working Links are updated

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4691)

### CODE REVIEW

**Comment by chenopis** [comment]
@SrinivasChilveri Actually, we don't link to the .md files. If you look at the [preview build](https://deploy-preview-4691--kubernetes-io-master-staging.netlify.com/docs/concepts/architecture/master-node-communication/), you'll see that using URLs to the .md files will cause a redirect. This is because the site renders the markdown into index.html files within a directory of the same name.

The update of the SSH Tunnels link, `/docs/concepts/architecture/master-node-communication/#ssh-tunnels`, is necessary, however. So thank you for that change.

Please remove the `.md` in the URLs and replace them w/ a slash. Thanks.

**Comment by SrinivasChilveri** [comment]
@chenopis ,I saw some links with .md file in the  https://github.com/kubernetes/kubernetes.github.io/blob/master/docs/concepts/architecture/nodes.md link address as  (https://git.k8s.io/community/contributors/design-proposals/architecture.md#the-kubernetes-node). any how i have changed as per your suggestion.

File: docs/concepts/architecture/master-node-communication.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ All communication paths from the cluster to the master terminate at the
 apiserver (none of the other master components are designed to expose remote
 services). In a typical deployment, the apiserver is configured to listen for
 remote connections on a secure HTTPS port (443) with one or more forms of
-client [authentication](/docs/admin/authentication/) enabled. One or more forms
+client [authentication](/docs/admin/authentication.md/) enabled. One or more forms
```
Comment (tengqm) [inline docs/concepts/architecture/master-node-communication.md]: don't think this change is needed

**Comment by chenopis** [comment]
@SrinivasChilveri That link goes to the GitHub repo (kubernetes/community), so it's fine to view documents directly in a repo using a path to the .md file. However, since the contents of this repo (kubernetes/kubernetes.github.io) are turned into the website kubernetes.io, links here are treated differently.

Thanks for making the changes!

**Pull request merged**